core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/user-factory.js

```diff
@@ -23,7 +23,7 @@ let e2eBlogAdmin = e2eBlogPostEditor = e2eGuestUser = require(
 /**
  * Global user instances that are created and can be reused again.
  */
-let superAdminInstance = null;
+let superAdminInstance = blogAdminInstance = null;
```

Comment:
blogAdminInstance shouldn't be a global here, I think. Only the super admin should be global. If you need to give the super admin blog admin permissions in order to do the setup, that's fine.

Reply:
Okay! tried to do a workaround

Reply:
A lot of time goes into creation of users, so i wanted to reduce this step by only initializing user if necessary and not everytime for every other test. This will greatly reduce our testing time for each spec in a suite.

Reply:
My previous point still stands: if you want to set up the test with new blog post editors, then please instead give the existing superadmin the permissions to do that, rather than creating and maintaining a new blog post admin user in this file.

Reply:
@seanlip I want to inform you that we cannot assign the `blog-editor` roles from the `/admin#/roles` page. only blog admins can assign the `blog-editor` roles from the `/blog-admin` page. that's why we need to have a blog admin so that we can assign a user with `blog-editor` role. also it would not be a good idea to assign blog admin role to the super admin.
<img width="722" alt="Screenshot 2023-02-20 at 12 51 04 PM" src="https://user-images.githubusercontent.com/91013793/220038820-413df1eb-fd0e-489d-a847-08779beceebc.png">

Reply:
@Shivkant-Chauhan You say "it would not be a good idea to assign blog admin role to the super admin", but this is exactly what I am proposing. Why wouldn't it be a good idea? There is no justification given.

Reply:
Sure, so in this case, the "hidden super admin" is used for setup. So why can't they be a blog admin as well, so that they can set up the roles for blog editors? That seems reasonable to me conceptually.

Reply:
@seanlip @Shivkant-Chauhan I am convinced with both the approaches. We can pick any one of them. In case of Sean's approach we will have to assign blog-admin role to the superadmin to assign blog post editor role whereas in case of Shivkant's approach we will create a new blogadmin user if not already created to assign the role of blog post editor. Both seems alright and i am flexible to any one of them.

Reply:
Yup, I still suggest that you set up exactly one "hidden" user and use that hidden user for all the necessary setup. You can set additional flags for "hasAdminUserBeenGivenBlogAdminPermissions" to avoid repeated steps, but overall this approach will be faster than creating a bunch of hidden users.

Reply:
@seanlip I just discovered we can't use `superAdmin` to do `blogAdmin` stuff. It's not because we can't, we can definitely. But the way we have structured our classes. `blog-post-admin-utils.js` and `super-admin-utils.js` are two seperate files. So utiliity methods of blog users can't be exercised by superadmin and vice-versa. So in all, we can't do this until and unless we compromise the structure of the inhertance of classes.

Reply:
Could you have the superadmin class inherit from the blog admin class?

Reply:
@ashish-patwal Filed https://github.com/oppia/oppia/issues/17416 (cf previous comment); you can refer to it when writing the TODO. Please feel free to leave a follow-up comment if I've missed something here.

Reply:
@seanlip Sure, I'll go with it then. I'll add the TODO comment as well. Thanks.